PWGHF/TableProducer/treeCreatorTccToD0D0Pi.cxx

```diff
@@ -0,0 +1,455 @@
+          const auto deltaMassD0D0Pi = massD0D0Pi - (massD01 + massD02);
+
+          if (deltaMassD0D0Pi > deltaMassCanMax) {
+            continue;
+          }
+
+          rowCandidateFull(
+            candidateD1.collisionId(),
+            candidateD1.pt(),
+            candidateD2.pt(),
+            trackPion.pt(),
+            candidateD1.pxProng0(),
+            candidateD1.pxProng1(),
+            candidateD1.pyProng0(),
+            candidateD1.pyProng1(),
+            candidateD1.pzProng0(),
+            candidateD1.pzProng1(),
+            candidateD2.pxProng0(),
+            candidateD2.pxProng1(),
+            candidateD2.pyProng0(),
+            candidateD2.pyProng1(),
+            candidateD2.pzProng0(),
+            candidateD2.pzProng1(),
+            trackPion.px(),
+            trackPion.py(),
+            trackPion.pz(),
+            candFlagD1,
+            candFlagD2,
+            massD01,
+            massD02,
+            deltaMassD01,
+            deltaMassD02,
+            massD0D0Pi,
+            deltaMassD0D0Pi,
+            candidateD1.eta(),
+            candidateD2.eta(),
+            trackPion.eta(),
+            candidateD1.phi(),
+            candidateD2.phi(),
+            trackPion.phi(),
+            yD1,
+            yD2,
+            trackPion.y(),
+            trackPion.tpcNSigmaPi(),
+            trackPion.tofNSigmaPi(),
+            mlScoresD1[0],
+            mlScoresD2[0],
+            candidateD1.decayLength(),
+            candidateD2.decayLength(),
+            candidateD1.cpa(),
+            candidateD2.cpa(),
+            trackPion.sign(),
+            trackPion.dcaXY(),
+            trackPion.dcaZ(),
+            trackPion.itsNCls(),
+            trackPion.tpcNClsCrossedRows(),
+            trackPion.tpcChi2NCl(),
+            cent);
+        } // end of loop track
+      } // end of loop second D0
+    } // end of loop first D0
+  }
+
+  void processDataWithML(CollisionsWCentMult const& collisions,
+                         SelectedCandidatesMl const& candidates,
+                         aod::TrackAssoc const& trackIndices,
+                         TracksWPid const& tracks,
+                         aod::BCs const& bcs)
+  {
+    for (const auto& collision : collisions) {
+      rowCandidateFullEvents.reserve(collisions.size());
+      fillEvent(collision, 0, collision.bc().runNumber());
+      auto thisCollId = collision.globalIndex();
+      auto candwD0ThisColl = candidates.sliceBy(candsD0PerCollisionWithMl, thisCollId);
+      if (candwD0ThisColl.size() <= 1)
+        continue; // only loop the collision that include at least 2 D candidates
+      auto trackIdsThisCollision = trackIndices.sliceBy(trackIndicesPerCollision, thisCollId);
+      runCandCreatorData<aod::hf_cand::VertexerType::DCAFitter>(collision, candwD0ThisColl, trackIdsThisCollision, tracks, bcs);
+    }
+  }
+  PROCESS_SWITCH(HfTreeCreatorTccToD0D0Pi, processDataWithML, "Process data with DCAFitterN", true);
+};
```

Comment:
What other process functions do you plan to add?

Reply:
> What other process functions do you plan to add?

for the moment, I would keep the task as much as simple, so only one process. As we said, it is a exploratory study